x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/telemetry/InferenceAPMStats.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.telemetry;
+
+import org.elasticsearch.inference.Model;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Objects;
+
+public class InferenceAPMStats extends InferenceStats {
+
+    private final LongCounter inferenceAPMRequestCounter;
+
+    public InferenceAPMStats(Model model, MeterRegistry meterRegistry) {
+        super(model);
+        this.inferenceAPMRequestCounter = meterRegistry.registerLongCounter(
+            counterName(),
+            "Inference API request counts for a particular service, task type, model ID",
+            "operations"
+        );
+    }
+
+    String counterName() {
+        StringBuilder builder = new StringBuilder();
```

Comment:
Rather than a counter per `(service, task type, model id)` combination these could be [dimensions](https://github.com/elastic/elasticsearch/blob/main/modules/apm/METERING.md#the-use-of-attributes-aka-dimensions) of a single counter `es.inference.requests.count`. These fields would be passed as the in the attributes map `void LongCounter::incrementBy(long inc, Map<String, Object> attributes)`